### PR TITLE
Add Pastebin.com translator

### PR DESCRIPTION
**Pull request by febrezo**
It processes both profiles and pastes and stores the website as an attachment. Test cases included.

### CODE REVIEW

File: Pastebin.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
+{
+	"translatorID": "1d1a51d4-60bf-44b8-ae28-8b154d1ed721",
+	"label": "Pastebin",
+	"creator": "febrezo",
+	"target": "https?://(www.)?pastebin.com",
```
Comment (adam3smith) [inline Pastebin.js]: escape periods and make more restrictive: `^https?://(www\\.)?pastebin\\.com/`

File: Pastebin.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 1,
```
Comment (adam3smith) [inline Pastebin.js]: 100

File: Pastebin.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+function getType(url) {
+	if ( url.indexOf('/u/')>-1 ) {
+		Zotero.debug("This is a profile...");
+		return "profile";
+	} else  	{
+		Zotero.debug("This is a paste");
+		return "paste";
+	} 
+}
+
```
Comment (adam3smith) [inline Pastebin.js]: Note sure what this is, but seems like it should be deleted.

File: Pastebin.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+		"rights": "",
+		"extra": "",
+		"tags": [],
+		"collections": [],
+		"relations": {}
+	}
+*/
+
+function doWeb(doc, url) {
+	Zotero.debug("=================================");
```
Comment (adam3smith) [inline Pastebin.js]: delete.

File: Pastebin.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	
+	resourceType = detectWeb(doc, url);
```
Comment (adam3smith) [inline Pastebin.js]: missing `var`

File: Pastebin.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+		newItem.date = date;	
+	} 
+	else if (newItem.websiteType == "profile")  {
+		var author = url.substring(url.lastIndexOf('/'));	
+		newItem.creators = [Zotero.Utilities.cleanAuthor(author, "author", false)];
+	}
+	
+	// Adding the attachment
+	newItem.attachments.push({
+		title: title,
```
Comment (adam3smith) [inline Pastebin.js]: don't use title for attachment titles. Should be "Pastebin Snapshot"

File: Pastebin.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+		mimeType: "text/html",
+		url: url
+	});	
+	
+	newItem.complete();
+ 
+	Zotero.debug("=================================");
```
Comment (adam3smith) [inline Pastebin.js]: delete

**Comment by zuphilip** [comment]
I am a little skeptical. Are there really enough metadata on these pages to come up with a reasonable translator?

**Comment by febrezo** [comment]
@adam3smith This was my first pull request so thanks a lot for the feedback to improve further translators. I'm working on the comments this morning.

@zuphilip In my work, I have to cite pastes because many leaks and manifestos are sometimes posted there. The automatic extraction of metadata of the default wrappers leads to inconsistent citations, (missing fields it is unable to extract the profile that posted the paste e. g.), date and so on. This is relevant for us because many pastes are sometimes deleted and Zotero allows us to cite the Snapshot using well-formed standard citation rules.

File: Pastebin.js (hunk 3 above)
Comment (febrezo) [inline Pastebin.js]: In Pastebin you can generally find two type of `webpages`: pastes themselves and, for the registered profiles, profile pages with some extra information about the profile. Metadata is extracted slightly differently in both pages so I coded this simple function to find out what type of page the user is viewing. Anyway, I can include that info inside doWeb.

**Comment by adam3smith** [comment]
@dstillman -- I think @febrezo 's rationale for adding these makes sense and is well within the spirit of what zotero is designed for and am inclined to take them, but I do see why @zuphilip is skeptical, given the minimal metadata. What do you think?

**Comment by febrezo** [comment]
I'll accept your decision whatever it is. Anyway, I'll provide an an example of what this translator does in comparison with the default one to so why I feel that it improves the current capabilities:
```
[
	{
		"id": "http://zotero.org/users/2557305/items/I9ZNZQH3",
		"type": "webpage",
		"title": "Bash, basic parameters processing",
		"container-title": "Pastebin.com",
		"genre": "Paste Site",
		"URL": "http://pastebin.com/FuzVdRiJ",
		"author": [
			{
				"family": "febrezo",
				"given": ""
			}
		],
		"issued": {
			"date-parts": [
				[
					"2012",
					5,
					27
				]
			]
		},
		"accessed": {
			"date-parts": [
				[
					"2016",
					9,
					3
				]
			]
		}
	},
	{
		"id": "http://zotero.org/users/2557305/items/DPRA58R4",
		"type": "webpage",
		"title": "[Bash] Bash, basic parameters processing - Pastebin.com",
		"URL": "http://pastebin.com/FuzVdRiJ",
		"accessed": {
			"date-parts": [
				[
					"2016",
					9,
					3
				]
			]
		}
	}
]
```

**Comment by dstillman** [comment]
Yeah, this is fine with me.

File: Pastebin.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+	You should have received a copy of the Affero GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+function detectWeb(doc, url) {
+	return "webpage";
+}
+
+// More item types available at: <http://gsl-nagoya-u.net/http/pub/csl-fields/index.html>
```
Comment (adam3smith) [inline Pastebin.js]: Please delete all the commented out bit from line 38-65.

File: Pastebin.js [hunk 9]
```diff
@@ -0,0 +1,226 @@
+{
+	"translatorID": "1d1a51d4-60bf-44b8-ae28-8b154d1ed721",
+	"label": "Pastebin",
+	"creator": "febrezo",
+	"target": "https?://pastebin\\.com",
```
Comment (adam3smith) [inline Pastebin.js]: put a ^ at the beginning here. That greatly enhances efficiency.

**Comment by adam3smith** [comment]
OK, two remaining nits, then this is good to go

**Comment by adam3smith** [comment]
Great, thanks!

**Pull request merged**